### PR TITLE
Add scaffolding command for quickly generating blocks

### PR DESCRIPTION
**Pull request by fabiankaegy**
### Description of the Change

This PR adds a new `scaffold:block` command to the 10up starter theme which allows anyone to easily scaffold out a new block. It is based on the `@wordpress/create-block` package with custom block templates that are bundled in the scaffold itself. 

Closes #

### How to test the Change

To test this you can run `npm run scaffold:block -w themes/10up-theme` in the `wp-content` folder. This will run through a few prompts and generate the block for you. 

### Changelog Entry

> Added - Block Scaffolding command to easily build out new blocks

### Credits

Props @fabiankaegy 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by fabiankaegy** [comment]
@nicholasio I'd love to get your take on this. Do you think this should live in the scaffold here or in toolkit. 

My reason for including it in the scaffold is that I want project teams to be able to modify the starter block templates to their linings. Which would be more difficult in toolkit since it doesn't get exposed. 

However if it were in toolkit it would be easier to ship updates to the scaffolding tool compared to here where essentially it gets stale once the project is scaffolded unless someone manually updates it.

File: bin/create-block-template/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+const { join } = require('path');
+
+module.exports = {
+	defaultValues: {
+		slug: 'example-block',
```
Comment (darylldoyle) [inline bin/create-block-template/index.js:8]: **question (non-blocking):** Should this by dynamically set by the script?

File: bin/create-block-template/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		category: 'text',
+		title: 'Example Block',
+		description: 'Example Block',
```
Comment (darylldoyle) [inline bin/create-block-template/index.js:11]: question (non-blocking): Should these by dynamically set by the script?

File: bin/create-block-template/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		attributes: {},
+		supports: {
+			html: false,
+		},
+		customBlockJSON: {
```
Comment (darylldoyle) [inline bin/create-block-template/index.js:16]: **question (non-blocking):** Maybe this is ignorance on my part, but what does this property do?

**Review by darylldoyle — COMMENTED** [review]
I love this! Some way to do this has always been at the top of my mind. 

I've left a couple of questions, but once those are answered it looks good!

**Comment by ivanlopez** [comment]
@fabiankaegy If we were to add it to the toolkit, we could also create a command to publish the templates into the project to be customized, like Laravel does here: https://laravel.com/docs/10.x/artisan#stub-customization.

File: bin/create-block-template/index.js (hunk 1 above)
Comment (fabiankaegy) [reply bin/create-block-template/index.js:8]: Yeah :) This value here only sets the default if the user doesn't provide one in the CLI prompts.

(See attached Video)


https://github.com/10up/wp-scaffold/assets/20684594/fe0562ed-c37b-417d-a077-bd3183f298c1

File: bin/create-block-template/index.js (hunk 2 above)
Comment (fabiankaegy) [reply bin/create-block-template/index.js:11]: Same answer as here: https://github.com/10up/wp-scaffold/pull/195#discussion_r1321655475

File: bin/create-block-template/index.js (hunk 3 above)
Comment (fabiankaegy) [reply bin/create-block-template/index.js:16]: https://github.com/WordPress/gutenberg/blob/trunk/packages/create-block/docs/external-template.md#defaultvalues

> customBlockJSON (no default) - allows definition of additional properties for the generated block.json file.

This allows us to add any other additional properties to the `block.json` files generated :)

**Comment by fabiankaegy** [comment]
@ivanlopez 👀 That does sound like an interesting solution that would cover both cases!

**Comment by nicholasio** [comment]
@fabiankaegy Since the actual logic to create scaffold the block is handled by `@wordpress/create-block` I don't think there's much value in moving this to toolkit as it would essentially wrap `@wordpress/create-block`.  One thing that toolkit could do it let projects override the default template, so if nothing is provided at the project level it would use the one shipped with toolkit.

With that said, it's probably fine to ship this directly in the scaffold for now.

**Comment by fabiankaegy** [comment]
Okay, in that case, let's keep it simple for now. 

We can ship this as is and iterate on it if we find we want more flexibility in the future

**Pull request merged**